modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/input_move_boxes.js

```diff
@@ -271,32 +282,28 @@ AUI.add(
 				_toggleBtnMove(event) {
 					var instance = this;
 
-					var contentBox = instance.get('contentBox');
+					var sourceBox = event.target;
 
-					var moveBtnLeft = contentBox.one('.move-left');
-					var moveBtnRight = contentBox.one('.move-right');
+					var selectedOptions = sourceBox
+						.get('options')
+						.getDOMNodes()
+						.filter((option) => option.selected);
 
-					var target = event.target;
+					var direction =
+						sourceBox === instance._rightBox ? 'left' : 'right';
 
-					if (moveBtnLeft && moveBtnRight && target) {
-						var btnDisabledLeft = true;
-						var btnDisabledRight = true;
+					var destinationBox = instance[`_${direction}Box`];
+					var destinationBoxMaxItems = instance.get(
+						`${direction}BoxMaxItems`
+					);
```

Comment:
Nice simplification, although it is tempting to do something a bit more dumb/explicit in the name of grepability (ie. so that if I grep for `leftBoxMaxItems` I actually find the usage here):

```js
if (sourceBox === instance._rightBox) {
    var destinationBox = instance._leftBox;
    var maxItems = instance.get('leftBoxMaxItems');
    var button = contentBox.one('.move-left');
}
else {
    var destinationBox = instance._rightBox;
    var maxItems = instance.get('rightBoxMaxItems');
    var button = contentBox.one('.move-right');
}
```

Reply:
That's what I did originally... but you said BRANCHES ARE FOR LOSERS AND I'M NO LOSER!! 💪

Reply:
Was also worried about what would happen when we let/const the hell out of this 😂

Reply:
PR is clean, and I'm fine amending like you suggest... just let me know

Reply:
I'm also fine either way. Sadly, you cannot have your cake and eat it too, so you must choose one.